cachito/workers/pkg_managers/gomod.py

```diff
@@ -316,6 +311,82 @@ def resolve_gomod(app_source_path, request, dep_replacements=None, git_dir_path=
         return {"module": module, "module_deps": module_level_deps, "packages": packages}
 
 
+def _should_vendor_deps(flags: List[str], app_dir: str, strict: bool) -> Tuple[bool, bool]:
+    """
+    Determine if Cachito should vendor dependencies and if it is allowed to make changes.
+
+    This is based on the presence of flags:
+    - gomod-vendor-check => should vendor, can only make changes if vendor dir does not exist
+    - gomod-vendor => should vendor, can make changes
+
+    :param flags: flags from the Cachito request
+    :param app_dir: absolute path to the app directory
+    :param strict: fail the request if the vendor dir is present but the flags are not?
+    :return: (should vendor: bool, allowed to make changes: bool)
+    :raise ValidationError: if the vendor dir is present, the flags are not and we are strict
+    """
+    vendor = Path(app_dir) / "vendor"
+
+    if "gomod-vendor-check" in flags:
+        return True, not vendor.exists()
+    elif "gomod-vendor" in flags:
+        return True, True
+
+    if strict and vendor.is_dir():
+        raise ValidationError(
+            'The "gomod-vendor" or "gomod-vendor-check" flag must be set when your repository has '
+            "vendored dependencies."
+        )
+
+    return False, False
+
+
+def _vendor_deps(run_params: dict, can_make_changes: bool, git_dir: str):
+    """
+    Vendor golang dependencies.
+
+    If Cachito is not allowed to make changes, it will verify that the vendor directory already
+    contained the correct content.
+
+    :param run_params: common params for the subprocess calls to `go`
+    :param can_make_changes: is Cachito allowed to make changes?
+    :param git_dir: path to the repository root
+    :raise ValidationError: if vendor directory changed and Cachito is not allowed to make changes
+    """
+    log.info("Vendoring the gomod dependencies")
+    run_gomod_cmd(("go", "mod", "vendor"), run_params)
+    app_dir = run_params["cwd"]
+    if not can_make_changes and _vendor_changed(git_dir, app_dir):
```

Comment:
Can you help me understand what the purpose of variable `can_make_changes`?

Reply:
If `can_make_changes` is True, then it is OK for Cachito to change the content of the `vendor/` directory. Otherwise, Cachito has to check that calling `go mod vendor` did not change anything.